bin/Audio_Transcript_GUI.py

```diff
@@ -0,0 +1,319 @@
+
+    def transcription(self, out):
+        logging.debug("Transcribed text: %s" % out)
+        self.textboxTranscript.insertPlainText(out)
+        self.show()
+
+    def threadComplete(self):
+        logging.debug("File processed")
+
+    def progress(self, chunk):
+        logging.debug("Progress: %s" % chunk)
+        self.textboxTranscript.insertPlainText(chunk)
+        self.show()
+
+    def runDeepspeech(self, waveFile, progress_callback):
+        '''
+        Deepspeech will be run from this function
+        '''
+        logging.debug("Caught the wav file @: %s" % (waveFile))
+        logging.debug("Preparing Deepspeech...")
+
+        N_FEATURES = 26
+        N_CONTEXT = 9
+        BEAM_WIDTH = 500
+        LM_WEIGHT = 1.75
+        VALID_WORD_COUNT_WEIGHT = 1.00
+        WORD_COUNT_WEIGHT = 1.00
+
+        # Run deepspeech on the chunk that just completed VAD
+        # Go and fetch the models from the directory specified
+        if self.dirName:
+
+            models = glob.glob(self.dirName + "/*.pb")[0]
+            logging.debug("Found Model: %s" % models)
+
+            alphabet = glob.glob(self.dirName + "/alphabet.txt")[0]
+            logging.debug("Found Alphabet: %s" % alphabet)
+
+            lm = glob.glob(self.dirName + "/lm.binary")[0]
+            trie = glob.glob(self.dirName + "/trie")[0]
+            logging.debug("Found Language Model: %s" % lm)
+            logging.debug("Found Trie: %s" % trie)
+
+        else:
+            logging.critical("*****************************************************")
+            logging.critical("Model path not specified..")
+            logging.critical("You sure of what you're doing ?? ")
+            logginf.critical("Trying to fetch from present working directory.")
+            logging.critical("*****************************************************")
+            models = " ./models/output_graph.pb "
+            alphabet = " ./models/alphabet.txt "
+            lm = " ./models/lm.binary "
+            trie = " ./models/trie "
+
+        f = open(waveFile.rstrip(".wav") + ".txt", 'w')
+        logging.debug("Saving Transcript @: %s" % waveFile.rstrip(".wav") + ".txt")
+
+        audio, sample_rate = waveSplit.read_wave(waveFile)
+        vad = webrtcvad.Vad(1)
+        frames = waveSplit.frame_generator(30, audio, sample_rate)
+        frames = list(frames)
+        segments = waveSplit.vad_collector(sample_rate, 30, 300, vad, frames)
+
+        model_time = 0.0
+        lm_time = 0.0
+        inference_time = 0.0
+        audio_time = 0.0
+
+        for i, segment in enumerate(segments):
+            # Run VAD on the input file
+            audio = 'chunk-%002d.wav' % (i,)
+
+            logging.debug(' Writing %s' % (audio,))
+            waveSplit.write_wave(audio, segment, sample_rate)
+
+            # Find Model load time
+            model_load_start = timer()
+            ds = Model(models, N_FEATURES, N_CONTEXT, alphabet, BEAM_WIDTH)
```

Comment:
So you load a model each time? That seems overkill, and low.

Reply:
Will move it out of the loop.

Reply:
You might even want to change the way the model (models actually) is being handled

Reply:
Need a little help with this. I tested this and it worked perfectly fine.
What other change are you foreseeing ?

Reply:
Well, being able to properly select it

Reply:
It does that :)
The `models` is a user select-able option.
![screenshot from 2018-09-12 20-32-51](https://user-images.githubusercontent.com/4998776/45434035-157e1180-b6cb-11e8-872b-fd3e2fe6fd85.png)